packages/x-data-grid/src/hooks/features/filter/gridFilterState.ts

```diff
@@ -20,7 +26,7 @@ export interface GridFilterState {
    * If a row is not registered in this lookup, it is filtered.
    * This is the equivalent of the `visibleRowsLookup` if all the groups were expanded.
    */
-  filteredRowsLookup: Record<GridRowId, boolean>;
+  filteredRowsLookup: Record<GridRowId, false>;
```

Comment:
Since we don't store `true` values now, does it make sense to simplify the type further? It could be slightly better performance wise. (same for the visible rows, column visibility model, etc)
```suggestion
  unfilteredRows: Set<GridRowId>;
```

Reply:
Changing `Record` to `Set`/`Map` won't necessarily be faster – see https://github.com/mui/mui-x/pull/9120#discussion_r1215040795

Reply:
See https://github.com/mui/mui-x/pull/10068#issuecomment-1743654847 and https://github.com/mui/mui-x/pull/10068#discussion_r1355431915. I think the change is good and has potential for improvement, but I didn't have the bandwidth to investigate it more.